src/vs/workbench/contrib/chat/browser/chatMarkdownDecorationsRenderer.ts

```diff
@@ -284,47 +282,56 @@ export class ChatMarkdownDecorationsRenderer extends Disposable {
 }
 
 
-class InlineFileLinkWidget extends Disposable {
+class InlineAnchorWidget extends Disposable {
 
 	constructor(
 		element: HTMLAnchorElement,
-		location: Location | { uri: URI; range: undefined },
+		data: ContentRefData,
 		@IHoverService hoverService: IHoverService,
 		@IInstantiationService instantiationService: IInstantiationService,
 		@ILabelService labelService: ILabelService,
 		@ILanguageService languageService: ILanguageService,
 		@IModelService modelService: IModelService,
-		@IThemeService themeService: IThemeService,
 	) {
 		super();
 
-		element.classList.add('chat-inline-file-link-widget');
+		element.classList.add('chat-inline-anchor-widget', 'show-file-icons');
+		element.replaceChildren();
 
-		const fragment = location.range ? `${location.range.startLineNumber}-${location.range.endLineNumber}` : '';
-		element.setAttribute('data-href', location.uri.with({ fragment }).toString());
+		const resourceLabel = this._register(new IconLabel(element, { supportHighlights: false, supportIcons: true }));
 
-		const label = labelService.getUriLabel(location.uri, { relative: true });
-		const title = location.range ?
-			`${label}#${location.range.startLineNumber}-${location.range.endLineNumber}` :
-			label;
+		let location: { readonly uri: URI; readonly range?: IRange };
+		if (data.kind === 'symbol') {
+			location = data.symbol.location;
 
-		element.replaceChildren();
+			const icon = SymbolKinds.toIcon(data.symbol.kind);
+			resourceLabel.setLabel(`$(${icon.id}) ${data.symbol.name}`, undefined, {});
```

Comment:
I think you can also pass the icon as an option. I don't like using the `$()` syntax internally.